File: Daily Temperatures - Leetcode 739/Daily Temperatures - Leetcode 739.java
public class DailyTemperatures {
    public static int[] dailyTemperatures(int[] temperatures){
        int n = temperatures.length;
        int[] ans = new int[n];
        Stack<int[]> stack = new Stack<>();

        for (int i = 0; i < n; i++)
        {
            while (stack.length > 0 && stack.peek()[0] > temperatures[i])
            {
                int[] t = stack.pop();
                int stack_i = t[1];
                ans[stack_i] = i - stack_i;
            }

            stack,push(new int[]{temperatures[i], i});
        }

        return ans;
    }
}

File: Daily Temperatures - Leetcode 739/Daily Temperatures - Leetcode 739.ts
export const dailyTemperatures = (temperatures: number[]) => {
  const n = temperatures.length;
  const stack: number[][] = [];
  const ans: number[] = Array(n).fill(0);

  for (let i = 0; i < n; i++) {
    const t = stack[stack.length - 1];
    while (stack.length > 0 && t[0] < temperatures[i]) {
      const stack_i = stack.pop()![1];
      ans[stack_i] = i - stack_i;
    }
  }

  return ans;
};
